tests/Jira/ApiTest.php

```diff
@@ -25,4 +25,102 @@ public function testSetEndpointTrailingSlash()
         $api->setEndPoint($url);
         $this->assertEquals($url, $api->getEndpoint());
     }
+
+    /**
+     * Tests that the updateVersion call constructs the correct api call
+     */
+    public function testUpdateVersion()
+    {
+        $params = array(
+            'released' => true,
+            'releaseDate' => '2010-07-06',
+        );
+
+        // Stub the api method and keep the rest intact
+        /** @var Api|\PHPUnit_Framework_MockObject_MockObject $api */
+        $api = $this->getMockBuilder('\chobie\Jira\Api')->setMethods(array('api'))->disableOriginalConstructor()->getMock();
```

Comment:
You know that PHPUnitMocks isn't that powerful and has it's own limitations. Since PHPUnit 4.8 the built-in support for Prophecy was added and people are encouraged to use it for mock creation.

It also encourages good architecture but at least not allowing to partially mock objects (e.g. `api` method call gets mocked, while others are not as you did above).